bson/src/test/unit/org/bson/BinaryVectorTest.java

```diff
@@ -16,17 +16,96 @@
 
 package org.bson;
 
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
+import org.slf4j.LoggerFactory;
 
 import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class BinaryVectorTest {
 
+    private ListAppender<ILoggingEvent> logWatcher;
+
+    @BeforeEach
+    void setup() {
+        logWatcher = new ListAppender<>();
+        logWatcher.start();
+        ((Logger) LoggerFactory.getLogger("org.bson.BinaryVector")).addAppender(logWatcher);
+    }
+
+    @AfterEach
+    void teardown() {
+        ((Logger) LoggerFactory.getLogger("org.bson.BinaryVector")).detachAndStopAllAppenders();
+    }
+
+    @DisplayName("Treatment of non-zero ignored bits: 1. Encoding")
+    @Test
+    void encodingWithNonZeroIgnoredBits() {
+        // when
+        byte[] data = {(byte) 0b11111111};
+
+        // then
+        assertDoesNotThrow(()-> BinaryVector.packedBitVector(data, (byte) 7));
+        ILoggingEvent iLoggingEvent = logWatcher.list.get(0);
+        assertEquals(Level.WARN, iLoggingEvent.getLevel());
+        assertEquals("The last 7 padded bits should be zero in the final byte.", iLoggingEvent.getMessage());
+    }
+
+    @DisplayName("Treatment of non-zero ignored bits: 2. Decoding")
+    @Test
+    void decodingWithNonZeroIgnoredBits() {
+        // when
+        byte[] bytearray = {0x10, 0x07, (byte) 0xFF};
+        BsonBinary data = new BsonBinary((byte) 9, bytearray);
+
+        // then
+        assertDoesNotThrow(data::asVector);
+        ILoggingEvent iLoggingEvent = logWatcher.list.get(0);
+        assertEquals(Level.WARN, iLoggingEvent.getLevel());
+        assertEquals("The last 7 padded bits should be zero in the final byte.", iLoggingEvent.getMessage());
+    }
+
+    @Test
+    void testComparisonWithIgnoredBits() {
```

Comment:
[nit] For consistency with other test method names in this class.
```suggestion
    void shouldCompareVectorsWithIgnoredBits() {
```